pkg/tcp/proxy.go

```diff
@@ -90,15 +93,21 @@ func (p *Proxy) ServeTCP(conn WriteCloser) {
 }
 
 func (p Proxy) dialBackend() (*net.TCPConn, error) {
```

Comment:
Could you add a test, similar to `TestLookupAddress`, to make sure a source IP is obtained from a configured list?